host/tests/dsCompositeIn_L3_Tests/dsCompositeIn_test2_VerifyStatus_Callback.py

```diff
@@ -0,0 +1,112 @@
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import re
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from dsCompositeInHelperClass import dsCompositeInHelperClass
+
+class dsCompositeIn_test2_VerifyStatus_Callback(dsCompositeInHelperClass):
+    """
+    Test class to select the compositeIn port and verify the status of CompositeIn device
+
+    This class uses the `dscompositeInClass` to interact with the device's compositeIn ports,
+    downloading necessary test assets, selecting
+    CompositeIn ports, and performing verification of compositeIn output.
+    """
+    def __init__(self):
+        """
+        Initializes the test2_VerifyStatus_Callback test .
+
+        Args:
+            None.
+        """
+        self.testName  = "test2_VerifyStatus_Callback"
+        super().__init__(self.testName, '1')
+
+    #TODO: Current version supports only manual verification.
+    def CheckDeviceStatus(self, manual=False, port_type:str=0):
+        """
+        Verifies whether Composite Source Device connected or not
+
+        Args:
+            port_type (str) : CompositeIn port
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f"Check if CompositeIn source is connected to {port_type} and press Y:")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """This function will test and verify the active status callback
+
+        This function:
+        - Connect / disconnect the compositeIn port.
+        - Verify the status through callbacks
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        self.log.testStart("test2_VerifyStatus_Callback", '1')
+
+        # Initialize the dsCompositeIn module
+        self.testdsCompositeIn.initialise()
+
+        # Loop through the supported ports
+        for port in self.testdsCompositeIn.getSupportedPorts():
+            self.log.stepStart(f'Select {port} Port')
+
+            portstr = f"dsCOMPOSITE_IN_PORT_{port}"
+            result = self.CheckDeviceStatus(True, portstr)
+            self.log.stepResult(result,f'CompositeIn Device connected {result} on {portstr}')
+          
+            self.testdsCompositeIn.selectPort(portstr)
+            self.log.stepStart(f'Port Selcted {portstr}')
+                
+            status = self.testdsCompositeIn.getPortCallbackStatus()
+            if status != None and status[1] == portstr:
+                result = True
+                self.log.stepResult(result,f'Port Status ispresented:{status[0]} activeport:{status[1]} found in Callback')
```

Comment:
directly you can add True or False, instead of updating Result Variable

Reply:
done